### PR TITLE
py-rios: add v2.0.2

### PR DESCRIPTION
**Pull request by neilflood**
The recent release of numpy-2 exposed a few minor incompatibilities in rios. Version 2.0.2 fixes these, and a `conflicts` directive is added here to prevent earlier versions being used with numpy-2.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-rios/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-rios/package.py:36]: (not required) If possible, could you leave a comment above this with the PR fixing the issue in `rios`?

**Comment by alalazo** [comment]
Waiting for maintainers review before merging. Ping me in a week or so if everything stays silent.

**Review by gillins — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**
